common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/TraceTimeIntervalWarningService.java

```diff
@@ -85,16 +91,49 @@ private int saveCheckins(List<CheckIn> checkins, Function<ByteString, byte[]> id
     return numberOfInsertedTraceWarnings;
   }
 
+  /**
+   * Store the given checkin data as {@link TraceTimeIntervalWarning} entities for the Protected Reports. Returns the
+   * number of inserted entities, which is useful for the case where there might be conflicts with the table constraints
+   * during the db save operations.
+   */
+  @Transactional
+  public int saveCheckInProtectedReports(List<CheckInProtectedReport> allCheckins, Integer submissionTimestamp) {
+    int numberOfCheckInProtectedReports = 0;
+    for (CheckInProtectedReport checkInProtectedReports : allCheckins) {
+      boolean checkInProtectedInsertedSuccessfully = checkInProtectedReportsRepository
+          .saveDoNothingOnConflict(checkInProtectedReports.getLocationIdHash().toByteArray(),
+              checkInProtectedReports.getIv().toByteArray(),
+              checkInProtectedReports.getEncryptedCheckInRecord().toByteArray(),
+              submissionTimestamp);
+
+      if (checkInProtectedInsertedSuccessfully) {
+        numberOfCheckInProtectedReports++;
+      }
+    }
+    return numberOfCheckInProtectedReports;
```

Comment:
should we raise an additional log message (WARN?) when the list.size() != numberOfCheckInProtectedReports

Reply:
I agree, but maybe only if number of unencrypted is greater than zero? @AliceBilc can you add this?